tests/unit/instrumentation/test_prometheus.py

```diff
@@ -0,0 +1,119 @@
+        # `_total` suffix is stripped off the metric name for `Counters`, but needed for clarity.
+        assert repr(prometheus.karapace_http_requests_total) == "prometheus_client.metrics.Counter(karapace_http_requests)"
+        assert (
+            repr(prometheus.karapace_http_requests_duration_seconds)
+            == "prometheus_client.metrics.Histogram(karapace_http_requests_duration_seconds)"
+        )
+        assert (
+            repr(prometheus.karapace_http_requests_in_progress)
+            == "prometheus_client.metrics.Gauge(karapace_http_requests_in_progress)"
+        )
+
+    def test_setup_metrics(self, caplog: LogCaptureFixture, prometheus: PrometheusInstrumentation) -> None:
+        app = AsyncMock(spec=RestApp, app=AsyncMock(spec=aiohttp.web.Application))
+
+        with caplog.at_level(logging.INFO, logger="karapace.instrumentation.prometheus"):
+            prometheus.setup_metrics(app=app)
+
+            app.route.assert_called_once_with(
+                prometheus.METRICS_ENDPOINT_PATH,
+                callback=prometheus.serve_metrics,
+                method="GET",
+                schema_request=False,
+                with_request=False,
+                json_body=False,
+                auth=None,
+            )
+            app.app.middlewares.insert.assert_called_once_with(0, prometheus.http_request_metrics_middleware)
+            app.app.__setitem__.assert_has_calls(
+                [
+                    call(prometheus.karapace_http_requests_total, prometheus.karapace_http_requests_total),
+                    call(
+                        prometheus.karapace_http_requests_duration_seconds,
+                        prometheus.karapace_http_requests_duration_seconds,
+                    ),
+                    call(prometheus.karapace_http_requests_in_progress, prometheus.karapace_http_requests_in_progress),
+                ]
+            )
+            for log in caplog.records:
+                assert log.name == "karapace.instrumentation.prometheus"
+                assert log.levelname == "INFO"
+                assert log.message == "Setting up prometheus metrics"
+
+    @patch("karapace.instrumentation.prometheus.generate_latest")
+    async def test_serve_metrics(self, generate_latest: MagicMock, prometheus: PrometheusInstrumentation) -> None:
+        await prometheus.serve_metrics()
+        generate_latest.assert_called_once_with(prometheus.registry)
+
+    @patch("karapace.instrumentation.prometheus.time")
+    async def test_http_request_metrics_middleware(
+        self,
+        mock_time: MagicMock,
+        prometheus: PrometheusInstrumentation,
+    ) -> None:
+        mock_time.time.return_value = 10
+        request = AsyncMock(
+            spec=aiohttp.web.Request, app=AsyncMock(spec=aiohttp.web.Application), path="/path", method="GET"
+        )
+        handler = AsyncMock(spec=aiohttp.web.RequestHandler, return_value=MagicMock(status=200))
```

Comment:
**[issue]**
I need to assert that the `handler` is called with the request, i forgot this, will add.